istioctl/cmd/istiodconfig.go

```diff
@@ -0,0 +1,464 @@
+	return ss, nil
+}
+
+type ControlzClient struct {
+	baseURL    *url.URL
+	httpClient *http.Client
+}
+
+func (c *ControlzClient) GetScopes() ([]*ScopeInfo, error) {
+	var scopeInfos []*ScopeInfo
+	resp, err := c.httpClient.Get(c.baseURL.String())
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("request not successful %s", resp.Status)
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&scopeInfos)
+	if err != nil {
+		return nil, fmt.Errorf("cannot deserialize response %s", err)
+	}
+	return scopeInfos, nil
+}
+
+func (c *ControlzClient) PutScope(scope *ScopeInfo) error {
+	var jsonScopeInfo bytes.Buffer
+	err := json.NewEncoder(&jsonScopeInfo).Encode(scope)
+	if err != nil {
+		return fmt.Errorf("cannot serialize scope %+v", *scope)
+	}
+	req, err := http.NewRequest(http.MethodPut, c.baseURL.String()+"/"+scope.Name, &jsonScopeInfo)
+	if err != nil {
+		return err
+	}
+	defer req.Body.Close()
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusAccepted {
+		return fmt.Errorf("cannot update resource %s, got status %s", scope.Name, resp.Status)
+	}
+	return nil
+}
+
+func (c *ControlzClient) GetScope(scope string) (*ScopeInfo, error) {
+	var s ScopeInfo
+	resp, err := http.Get(c.baseURL.String() + "/" + scope)
+	if err != nil {
+		return &s, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return &s, fmt.Errorf("request not successful %s: ", resp.Status)
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&s)
+	if err != nil {
+		return &s, fmt.Errorf("cannot deserialize response: %s", err)
+	}
+	return &s, nil
+}
+
+var (
+	istiodLabelSelector = ""
+	istiodReset         = false
+)
+
+func istiodLogCmd() *cobra.Command {
+	var opts clioptions.ControlPlaneOptions
+	var outputLogLevel = ""
+	var stackTraceLevel = ""
+
+	// output format (yaml or short)
+	var outputFormat = "short"
+
+	logCmd := &cobra.Command{
+		Use:   "log [--level <scope>:<level>][--stack-trace-level <scope>:<level>]|[-r|--reset]|[--output|-o short(default)|yaml]",
+		Short: "Manage istiod logging.",
+		Long:  "Retrieve or update logging levels of istiod components.",
+		Example: ` # Retrieve information about istiod logging levels.
+  istioctl istiod-config log [<pod-name>]
+
+  # Update levels of the specified loggers.
+  istioctl istiod-config log --level ads:debug,authorization:debug
+
+  # Reset levels of all the loggers to default value (info).
+  istioctl istiod-config log -r
+`,
+		Aliases: []string{"o"},
+		Args: func(logCmd *cobra.Command, args []string) error {
+			if istiodReset && outputLogLevel != "" {
+				logCmd.Println(logCmd.UsageString())
+				return fmt.Errorf("--level cannot be combined with --reset")
+			}
+			if istiodReset && stackTraceLevel != "" {
+				logCmd.Println(logCmd.UsageString())
+				return fmt.Errorf("--stack-trace-level cannot be combined with --reset")
```

Comment:
Just throwing it out there, my 2c is that stack trace level is not very useful. Perfectly happy to include it, but I would be fine with removing it as well if we wanted to

Reply:
I can remove it if this adds complexity